fix: Change map to geo_equals

xmnlab opened the pull request:
This PR fix the problem with `geo_equals`.

Currently, `equals` operation is mapped to `geo_equals` and it is given problems as reported at #2197

Fix #2197

jreback commented on ibis/backends/tests/test_geospatial.py:
```diff
@@ -686,3 +686,45 @@ def test_geo_ops_smoke(backend, fn_expr):
+@pytest.mark.xfail_unsupported
+def test_geo_equals(backend, geo):
+    """
+    Test geo_equals/equals.
```
reference the issue in a comment. the Notes section is not needed here

jreback reviewed (changes requested):
tests are failing; also pls add a release note

xmnlab commented:
@jreback thanks for the review.

I added the release note and also changed 2 tests that were using `.equals`. to `geo_equals`.
In ibis, as I understood, `equals` usually returns a boolean, not an ibis expression.

I also updated the documentation about `.equals`

datapythonista commented on ibis/backends/postgres/tests/test_postgis.py:
```diff
@@ -149,9 +149,16 @@ def test_geo_disjoint(geotable):
+    assert expr.execute().all()
+
+    expr = geotable.geo_point.geo_equals(geotable.geo_linestring)
+    assert not expr.execute().all()
```
Do you want all them to be different? Then `any` should be better I think.

xmnlab replied:
that makes sense. I will change it now. thanks @datapythonista

datapythonista reviewed (approved):
lgtm

jreback commented:
thanks @xmnlab
